pkg/get/get_test.go

```diff
@@ -810,7 +812,18 @@ func Test_DownloadInletsctl(t *testing.T) {
 			t.Fatal(err)
 		}
 		if got != tc.url {
-			t.Fatalf("want: %s, got: %s", tc.url, got)
+			t.Errorf("for %s/%s, want: %q, but got: %q", tc.os, tc.arch, tc.url, got)
+		}
+		name, err := GetBinaryName(tool, tc.os, tc.arch, tc.version)
+		if err != nil {
+			t.Fatal(err)
+		}
+		// Remove the tgz and possible exe extensions from the url base name
```

Comment:
Hmm. this p

Reply:
Note that the changes that are needed to fix getting inletsctl only affect the Name in its BinaryTemplate.

This test, which fails prior to my changes and succeeds after, verifies that the Name in the BinaryTemplate is consistent with the base fileName in the URLTemplate.

The complication that makes this more difficult to explain is that the `.exe` extension is conditionally added in decompress.

Do you think `arkade get inletsctl` is not currently broken? tool-test.sh has two failures. Or that it is not fixed by my changes?

Reply:
Why are we doing `nameFromUrl := strings.Split(filepath.Base(tc.url), ".")[0]`?

It wasn't clear in isolation.

Now I think I get it.. you're trying to infer the name from the URL - but that may not always be the case.

We may want a new value in the test struct i.e. `wantBinaryName`?

Reply:
I started by adding a binary _base_ name to the test struct but that didn't seem to help, or at least not carry its weight, imo. It was just replicating fileName from the URL, without exe.

I don't think this can be tested more effectively without modeling what the tgz file contains and what the decompress function does.

In the case of inletsctl, maybe it's possible to eliminate one of the templates? That would eliminate the original problem of them being out of sync.